### ISSUE TITLE
[xna] [monogame] [csharp] Issues enabling Multiply and Screen blend modes

### ISSUE THREAD

**Issue by randrade23** [EsotericSoftware/spine-runtimes#2010]
I am trying to enable blend modes in the spine-monogame / spine-xna runtimes.

I checked the source code for SkeletonRenderer and noticed the following code:

https://github.com/EsotericSoftware/spine-runtimes/blob/d065c88ca8ccba5040b353ceaf3c9799add5e624/spine-xna/src/SkeletonRenderer.cs#L153-L158

That code is supposed to change the BlendState loaded onto the GraphicsDevice so that different blend modes can be used, however it is commented - presumably since XNA/MonoGame does not define Multiply and Screen blend modes natively, it only defines modes for premultiplied alpha, non-premultiplied alpha, and Additive blending.

So, I uncommented the code and added the following:

```cs
// set blend state according to the slot we are drawing
BlendState blendState = defaultBlendState;
switch (slot.Data.BlendMode)
{
        case BlendMode.Additive:
            blendState = BlendState.Additive;
            break;
        case BlendMode.Multiply:
            blendState = multiplyBlend;
            break;
        case BlendMode.Screen:
            blendState = screenBlend;
            break;
         default:
         case BlendMode.Normal:
            blendState = premultipliedAlpha ? BlendState.AlphaBlend : BlendState.NonPremultiplied;
            break;
}

if (device.BlendState != blend)
{
         End();
         device.BlendState = blend;
}
```

The code will check what blend mode is present on the current slot, and flush the GraphicsDevice with its current blend mode before changing over into the appropriate mode for the current slot.

I defined ```multiplyBlend``` and ```screenBlend``` as such:

```cs
BlendState multiplyBlend = new BlendState()
{
          ColorBlendFunction = BlendFunction.Add,
          ColorSourceBlend = Blend.Zero,
          ColorDestinationBlend = Blend.SourceColor,
};
BlendState screenBlend = new BlendState()
{
          ColorBlendFunction = BlendFunction.Add,
          ColorSourceBlend = Blend.One,
          ColorDestinationBlend = Blend.InverseSourceColor,
};
```

I based myself on the formulas available here: [Photoshop Blend Modes](https://community.monogame.net/t/playing-with-blendstate-vs-photoshop-blend-modes/6827)

In my testing, only the screen blend mode is working correctly - when rendering slots that use the multiply blend mode, they show as black, and I cannot figure out why.

**Comment by badlogic** [comment]
XNA/MonoGame boend modes are indeed a mystery. I originally went down the same path as you, with the same results. I'm unsure how to resolve this issue.

**Comment by randrade23** [comment]
I believe I have figured it out!

I got the blend modes definitions right - all that was missing was that the Spines had to be exported with premultiplied alpha enabled for the MonoGame blend states to work correctly. I'll submit a PR later or tomorrow :)

**Comment by HaraldCsaszar** [comment]
I'm very sorry that I missed this ticket earlier! @randrade23 please note that blend mode _multiply_ has recently been implemented in this commit, only on the 4.2-beta branch:
https://github.com/EsotericSoftware/spine-runtimes/commit/66392e8cba3441945461bf639b463d5b67939bf4
If you have code for _screen_ blend mode ready and working as well, we're very happy to integrate it of course!
